### ISSUE TITLE
py3: broken progress bar

### ISSUE THREAD

**Issue by rickysarraf** [rickysarraf/apt-offline#65]
Part of the bug (lonstanding cursor annoyance) got resolved with the py3 port but then the item count in the progress bar broke again. Needs investigation later.

```
Downloading http://security.debian.org/dists/testing/updates/non-free/i18n/Translation-en.xz                                                             
WARN: http://security.debian.org/dists/testing/updates/contrib/i18n/Translation-en_IN.gz - 404 - Not Found                                                             
WARN: Will still try with other package uris####] 100.0% of 74 MiB
http://security.debian.org/dists/testing/updates/non-free/binary-all/Packages.xz done                                                             
http://security.debian.org/dists/testing/updates/non-free/binary-i386/Packages.xz done                                                             
WARN: http://security.debian.org/dists/testing/updates/non-free/i18n/Translation-en_IN.xz - 404 - Not Found                                                             
WARN: Will still try with other package uris
WARN: http://security.debian.org/dists/testing/updates/non-free/i18n/Translation-en.xz - 404 - Not Found                                                             
WARN: Will still try with other package uris
WARN: http://security.debian.org/dists/testing/updates/contrib/i18n/Translation-en_IN.lzma - 404 - Not Found                                                             
WARN: Will still try with other package uris
WARN: http://security.debian.org/dists/testing/updates/non-free/i18n/Translation-en_IN.bz2 - 404 - Not Found                                                             
WARN: Will still try with other package uris
http://security.debian.org/dists/testing/updates/non-free/i18n/Translation-en.bz2 done                                                             
WARN: http://security.debian.org/dists/testing/updates/non-free/i18n/Translation-en_IN.gz - 404 - Not Found                                                             
WARN: Will still try with other package uris
WARN: http://security.debian.org/dists/testing/updates/non-free/i18n/Translation-en_IN.lzma - 404 - Not Found                                                             
WARN: Will still try with other package uris
222 / 192 items: [##############################] 100.0% of 74 MiB

WARN: The following files failed to be downloaded.
Not all errors are fatal. For eg. Translation files are not present on all mirrors.
WARN: http://debug.mirrors.debian.org/debian-debug/dists/unstable-debug/main/i18n/Translation-en_IN.lzma failed.
WARN: http://debug.mirrors.debian.org/debian-debug/dists/unstable-debug/contrib/i18n/Translation-en_IN.lzma failed.
WARN: http://debug.mirrors.debian.org/debian-debug/dists/unstable-debug/non-free/i18n/Translation-en_IN.lzma failed.
WARN: http://debug.mirrors.debian.org/debian-debug/dists/testing-debug/main/i18n/Translation-en_IN.lzma failed.
WARN: http://debug.mirrors.debian.org/debian-debug/dists/testing-debug/contrib/i18n/Translation-en_IN.lzma failed.
WARN: http://debug.mirrors.debian.org/debian-debug/dists/testing-debug/non-free/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/testing/main/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/testing/contrib/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/testing/non-free/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/unstable/main/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/unstable/contrib/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/unstable/non-free/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/experimental/main/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/experimental/contrib/i18n/Translation-en_IN.lzma failed.
WARN: http://deb.debian.org/debian/dists/experimental/non-free/i18n/Translation-en_IN.lzma failed.
WARN: http://security.debian.org/dists/testing/updates/main/i18n/Translation-en_IN.lzma failed.
WARN: http://security.debian.org/dists/testing/updates/contrib/i18n/Translation-en_IN.lzma failed.
WARN: http://security.debian.org/dists/testing/updates/non-free/i18n/Translation-en_IN.lzma failed.

Downloaded data to /tmp/set.zip
2017-05-22 / 01:52:32 ♒♒♒  ☺  
```

**Comment by rickysarraf** [comment]
Okay! For sure, the issue is only with apt's db updates only. Which must be happening when we retry through the possible archive formats supported by the remote server, until we give up.

As shown below, the progress bar does not corrupt when doing deb downloads, where there are no retries. In the below example, it has cycled through apt's cache as well and downloaded some .debs from the web, without any corruption to the progress bar's count.

```
rrs@learner:~$ apt-offline get /var/tmp/lxc/debtemplate.uris --bundle /var/tmp/lxc/debtemplate.zip --cache-dir /var/cache/apt/archives/ --threads 5

Fetching APT Data

WARNING: If you are on a slow connection, it is good to
WARNING: limit the number of threads to a low number like 2.
WARNING: Else higher number of threads executed could cause
WARNING: network congestion and timeouts.

libc-dev-bin found in cache                                                            
debianutils found in cache                                                            
base-files found in cache                                                            
bash found in cache                                                            
libc6-dev found in cache                                                            
Downloading linux-libc-dev - 1 MiB                                                             
libquadmath0 found in cache                                                            
libitm1 found in cache                                                            

....snipped......

diffstat found in cache                                                            
gcc found in cache                                                            
gnupg2 found in cache                                                            
libglib2.0-0 found in cache                                                            
libcroco3 found in cache                                                            
libisc-export160 found in cache                                                            
libdns-export162 found in cache                                                            
libicu57 found in cache                                                            
libxdelta2 found in cache                                                            
linux-libc-dev done                                                             
lintian found in cache                                                            
manpages-dev found in cache                                                            
psmisc found in cache                                                            
xdelta found in cache                                                            
xdelta3 found in cache                                                            
157 / 159 items: [##############################]  98.8% of 90 MiB
libssl1.1 done                                                             
libssl1.0.2 done                                                             
159 / 159 items: [##############################] 100.0% of 90 MiB
Downloaded data to /var/tmp/lxc/debtemplate.zip
2017-05-26 / 10:16:03 ♒♒♒  ☺  
rrs@learner:~$ 
```

**Comment by rickysarraf** [comment]
Fixed with f455cf575842c3c936258762ecb613c1daf0c8d6
